youtube_dl/extractor/mediasite.py

```diff
@@ -21,7 +21,7 @@
 
 
 class MediasiteIE(InfoExtractor):
-    _VALID_URL = r'(?xi)https?://[^/]+/Mediasite/Play/(?P<id>[0-9a-f]{32,34})(?P<query>\?[^#]+|)'
+    _VALID_URL = r'(?xi)https?://[^/]+/Mediasite/(Play|Showcase/livebroadcast/Presentation)/(?P<id>[0-9a-f]{32,34})(?P<query>\?[^#]+|)'
```

Comment:
Don't capture groups you don't use. Add only matching test.

Reply:
I'm having some issues due to an unrelated bug/change in mediasite?

```ERROR: [Mediasite]  is not a supported codec; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/media/data/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 793, in extract_info
    ie_result = ie.extract(url)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 508, in extract
    ie_result = self._real_extract(url)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/mediasite.py", line 183, in _real_extract
    fatal=False))
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 2279, in _extract_ism_formats
    return self._parse_ism_formats(ism_doc, urlh.geturl(), ism_id)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 2306, in _parse_ism_formats
    self.report_warning('%s is not a supported codec' % fourcc)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 888, in report_warning
    '[%s] %s%s' % (self.IE_NAME, idstr, msg))
  File "/media/data/Projects/youtube-dl/test/helper.py", line 245, in _report_warning
    real_warning(w)
  File "test/test_download.py", line 52, in report_warning
    raise ExtractorError(message)
youtube_dl.utils.ExtractorError: [Mediasite]  is not a supported codec; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

This occurs for all other tests too.

Reply:
Add `only_matching` test.

Reply:
> Add `only_matching` test.

I'm sorry I don't really understand what you mean...